compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1788,7 +1788,23 @@ impl<'a> Parser<'a> {
                     }
                 }
             } else {
-                self.expected_ident_found()
+                let mut err = self.expected_ident_found();
+                if let Some((ident, _)) = self.token.ident() && ident.as_str() == "let" {
```

Comment:
This probably could've been `self.token.is_keyword(kw::Let)`, see [`Token::is_keyword`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_ast/token/struct.Token.html#method.is_keyword)

Reply:
Alternatively, this could've used [`Parser::eat_keyword_noexpect`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_parse/parser/struct.Parser.html#method.eat_keyword_noexpect) to avoid the bump call below

Reply:
Awesome, will send an improvement PR later on. 👍🏻